config-ui/src/components/Nav.jsx

```diff
@@ -26,11 +26,41 @@ import {
 import '@/styles/nav.scss'
 import { ReactComponent as SlackIcon } from '@/images/slack-mark-monochrome-black.svg'
 import { ReactComponent as SlackLogo } from '@/images/slack-rgb.svg'
+import UIContext from '../store/UIContext'
+import useWindowSize from '../hooks/useWIndowSize'
+
 
 const Nav = () => {
+  const uiContext = useContext(UIContext);
+  const [menuClass, setMenuClass] = useState("navbarMenuButton");
+  const size = useWindowSize();
+
+  const toggleSidebarOpen = (open) => {
```

Comment:
I think this could be refactored to be more declarative and drop the need for using the `IF` and `return` statements.
```suggestion
  const toggleSidebarOpen = (open) => {
  uiContext.changeSidebarVisibility(open)
  setMenuClass(open ? 'navbarMenuButtonSidebarOpened' : 'navbarMenuButton'
 }
```